src/ukify/ukify.py

```diff
@@ -918,6 +928,115 @@ def make_uki(opts):
 
     print(f"Wrote {'signed' if sign_args_present else 'unsigned'} {opts.output}")
 
+def sign_uki(opts):
+    """
+    Unpack UKI, replace cmdline, measure and sign PCR, then repack and sign it.
+    We have to repack as PCR signature should be before the linux section.
+    """
+    if not opts.sb_key and not opts.sb_cert_name:
+        raise ValueError("Signing arguments required to sign UKI")
+
+    pe = pefile.PE(opts.uki, fast_load=True)
+
+    uki = UKI(opts.uki)
+    linux_section = None
+    sign_kernel = opts.sign_kernel
+
+    measured_sections = [
+        '.osrel',
+        '.cmdline',
+        '.dtb',
+        '.uname',
+        '.splash',
+        '.pcrpkey',
+        '.initrd',
+        '.ucode',
+        '.linux'
+    ]
+    additional_sections = []
+    for section in pe.sections:
+        name = section.Name.rstrip(b"\x00").decode()
+
+        if name not in measured_sections:
+            continue
+
+        if name == ".cmdline" and opts.cmdline:
+            cmdline = opts.cmdline
+            if opts.append_cmdline:
+                cmdline = section.get_data().rstrip(b"\x00").decode() + " " + cmdline
+            uki_section = Section.create(name, cmdline, measure=True)
+            additional_sections += [uki_section]
+        else:
+            uki_section = Section.create(name, section.get_data(), measure=True)
+
+        if name == ".linux":
+            linux_section = uki_section
+        else:
+            uki.add_section(uki_section)
+
+    assert linux_section is not None
```

Comment:
Is the `assert` here to narrow the type and get mypy to stop complaining about something? If not, I'd suggest an 
```suggestion
    if linux_section is None:
        raise ValueError(…)
```
(or other error that's a better fit) here, since asserts can be compiled out with a simple environment variable.